js/src/wp-data/selectors.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Get terms for a taxonomy.
+ *
+ * @param {object} state    State object.
+ * @param {string} taxonomy The taxonomy's slug.
+ *
+ * @returns {object} Action object.
+ */
+export const getTerms = ( state, taxonomy ) => {
+	return state.terms[ taxonomy ];
+};
+
+/**
+ * Get the terms for the current post type.
+ *
+ * @param {object} state State object.
+ *
+ * @returns {object} Action object.
+ */
+export const getTaxonomies = ( state ) => {
+	return state.taxonomies;
+};
+
+/**
+ * Get the taxonomy for the given taxonomy's slug.
+ *
+ * @param {object} state        State object.
+ * @param {string} taxonomySlug Taxonomy slug.
+ *
+ * @returns {object} Taxonomy object.
+ */
+export const getTaxonomy = ( state, taxonomySlug ) => {
+	if ( ! state.taxonomies ) {
```

Comment:
So why is this?

Reply:
Because if the taxonomies have not been retrieved jet taxonomies will be `null`, leading to an error. If the taxonomies have been retrieved `taxonomies` will be an object, and in the worst case `state.taxonomies[ taxonomySlug ]` will return undefined, but will never cause a javascript error.